folly/Random.cpp

```diff
@@ -26,16 +26,33 @@
 #include <folly/File.h>
 #include <folly/FileUtil.h>
 
+#ifdef _MSC_VER
+#include <wincrypt.h>
+#endif
+
 namespace folly {
 
 namespace {
 
 void readRandomDevice(void* data, size_t size) {
+#ifdef _MSC_VER
+  static bool initialized = false;
```

Comment:
I believe my thought process on that might have come from the way `CryptReleaseContext` is documented. Either way, I've switch this to a proper mutex instead.